### DISCUSSION TITLE
How to add vite ssr options in quasar config?

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#15217] [CLI - SSR mode] [answered]
Trying to figure out how to add vite ssr options in config: https://vitejs.dev/config/ssr-options.html
 have tried:
 ```
 extendViteConf (viteConf) {
        viteConf.ssr = { noExternal: ['@fortawesome/*'] }
      },
```
and 
 ```
 extendViteConf (viteConf) {
        viteConf.build.ssr = { noExternal: ['@fortawesome/*'] }
      },
```
and
```
 extendViteConf (viteConf) {
        viteConf.ssr.noExternal: ['@fortawesome/*']
      },
```
also tried to add it to the ssr block, which also failed:
```
ssr: {
      noExternal: ['@fortawesome/*'],
```      
all to no avail. Anyone know the correct syntax?

**Comment by yusufkandemir** [comment, marked as answer]
Hi. You have to remember that your code is not the only place extending the Vite config. Vite plugins, AEs, Quasar CLI, etc. all can extend the config. So, you should safely merge the Vite config to avoid overriding the others:
```js
const { mergeConfig } = require('vite')
// ...
extendViteConf (viteConf) {
  viteConf = mergeConfig(viteConf, {
    ssr: {
      noExternal: ['@fortawesome/*']
    }
  });
},
```

Also, use Vite v2 documentation to not get misled by additions/changes in Vite 3&4. https://v2.vitejs.dev/config/#ssr-options

You can run `quasar inspect` to ensure the config is merged correctly. If it got merged correctly but still doesn't work, then there are two **possibilities**:
1. `noExternal: ['@fortawesome/*']` form might not be supported, so you can try using a RegEx instead (`noExternal: [/@fortawesome\/.*/]`).
2. The `noExternal` is not what you are looking for. In that case, I'd suggest sharing the error, wrong behavior, expected behavior, etc., and reproduction to get further help.

**Reply by ssuess** [reply]
Thanks!
